src/sections/create-dataset/CreateDatasetForm.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { FormEvent, ChangeEvent, useState } from 'react'
+import { Alert, Button, Col, Form, Row } from '@iqss/dataverse-design-system'
+import { useTranslation } from 'react-i18next'
+import { RequiredFieldText } from '../../components/form/RequiredFieldText/RequiredFieldText'
+import { SeparationLine } from '../../components/layout/SeparationLine/SeparationLine'
+import { useFormContext } from './CreateDatasetContext'
+import {
+  CreateDatasetFormFields,
+  FormValidationService,
+  FormValidationResult
+} from '../../dataset/domain/useCases/createDataset'
+import { FormSubmissionService } from '../../dataset/domain/useCases/createDataset'
+
+type CreateDatasetFormProps = {
+  formValidationService: FormValidationService
+  formSubmissionService: FormSubmissionService
+}
+
+enum SubmissionStatus {
+  Unsubmitted,
+  Submitted,
+  Errored
+}
+
+export const CreateDatasetFormPresenter: React.FC<CreateDatasetFormProps> = ({
+  formValidationService,
+  formSubmissionService
+}) => {
+  const { formState, updateFormState } = useFormContext()
+  const [submissionStatus, setSubmissionStatus] = React.useState<SubmissionStatus>(
+    SubmissionStatus.Unsubmitted
+  )
+  const [isSubmitting, setIsSubmitting] = React.useState<boolean>(false)
+  const [formErrors, setFormErrors] = useState<
+    Record<keyof CreateDatasetFormFields, string | undefined>
+  >({
+    createDatasetTitle: undefined
+  })
+  const { t } = useTranslation('createDataset')
+
+  const handleCreateDatasetFieldChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target
+    updateFormState({ ...formState, [name]: value })
+  }
+
+  const handleCreateDatasetSubmit = (event: FormEvent<HTMLFormElement>) => {
```

Comment:
Move to useFormContext
Expose the form states to the form for reuse within more forms